docs/spec/exceptions.rst

```diff
@@ -0,0 +1,62 @@
+Exceptions
+==========
+
+Some type checking behaviors, such as type narrowing and reachability analysis,
+require a type checker to understand code flow. Code flow normally proceeds
+from one statement to the next, but some statements such as ``for``, ``while``
+and ``return`` can change the code flow. Similarly, ``try``/``except``/``finally``
+statements affect code flow and therefore can affect type evaluation. For example::
+
+    x = None
+    try:
+        some_function()
+        x = 1
+    except:
```

Comment:
Do we really want to use a bare `except` here? It's hard to prove that *no* exception can be raised, which seems necessary to prove that we always reach `x = 1`. It might be better to use a concrete exception (e.g. `ValueError` or `RuntimeError`) in the example, to make it believable that the type checker can carry out the needed analysis -- and to avoid using constructs that are not recommended as best practice.